packages/protocol/src/.indexignore

```diff
@@ -0,0 +1,2 @@
+utils/test-util.ts
+di
```

Comment:
Nit: Could you please add some comments above each line as to why we are doing this? I don't think it is very obvious, e.g.,:

```suggestion
# do not export test-util in index to avoid a necessary dependency on test frameworks such as chai
# adopters can still use the functionality by importing it through the complete path
utils/test-util.ts
# do not export the DI utilities to avoid a necessary dependency on reflect-metadata
di
```